client/starwhale/core/model/model.py

```diff
@@ -945,43 +948,80 @@ def diff(self, compare_uri: Resource) -> t.Dict[str, t.Any]:
+        model_info = ModelInfoVo(**model_uri.info())
+
+        # TODO: When we have a better way to handle this, we can remove this.
+        overwrite_specs_str = ""
+        if overwrite_specs:
+            # overwrite specs format example:
+            # {"handler_name": {"replicas": 1, "resources": {"memory": "1GiB"}}}
+            job_specs = {
+                s.name: s
+                for s in model_info.version_info.step_specs
+                if s.job_name == run_handler
+            }
+
+            for handler_name, overwrite_spec in overwrite_specs.items():
+                if handler_name not in job_specs:
+                    raise ValueError(f"run_handler {handler_name} not found")
+
+                spec = job_specs[handler_name]
+                if "replicas" in overwrite_spec:
+                    spec.replicas = overwrite_spec["replicas"]
+
+                if "resources" in overwrite_spec:
+                    spec.resources = Handler._transform_resource(
+                        overwrite_spec["resources"]
+                    )
+
+            # Server api only accepts spec yaml dump str format.
+            # When we config overwrite_specs, we should not config run_handler, if not, server will raise Exception.
+            overwrite_specs_str = yaml.dump([s.dict() for s in job_specs.values()])
+            if overwrite_specs_str:
+                run_handler = None
+
+        dataset_ids = []
+        for _uri in dataset_uris or []:
+            if isinstance(_uri, str):
+                _uri = Resource(_uri, ResourceType.dataset)
+            dataset_ids.append(_uri.info()["versionId"])
+
+        if runtime_uri and isinstance(runtime_uri, str):
+            runtime_uri = Resource(runtime_uri, ResourceType.runtime)
+        runtime_id = (
+            runtime_uri.info()["versionId"] if isinstance(runtime_uri, Resource) else ""
         )
-        runtime_version_url = ""
-        if _runtime_uri:
-            runtime_version_url = (
-                _runtime_uri.info().get("versionId") or _runtime_uri.version
-            )
 
-        req = JobRequest(
-            model_version_url=_model_uri.info().get("versionId") or _model_uri.version,
-            dataset_version_urls=",".join(
-                [str(i.info().get("versionId") or i.version) for i in _dataset_uris]
-            ),
-            runtime_version_url=runtime_version_url,
+        kwargs: t.Dict = dict(
+            model_version_url=model_info.version_id,
+            dataset_version_urls=",".join(dataset_ids),
+            runtime_version_url=runtime_id,
             resource_pool=resource_pool,
-            handler=run_handler,
+            dev_mode=dev_mode,
+            dev_password=dev_mode_password,
         )
+
+        if ttl > 0:
+            # if want to live forever, do not set this field for server api
+            kwargs["time_to_live_in_sec"] = ttl
+
+        if run_handler is None:
```

Comment:
Replace with 'if run_handler:'? so None or '' would be covered.

Reply:
updated